### ISSUE TITLE
2020-11-15-MNIST.ipynb

### ISSUE THREAD

**Issue by tylertaewook** [Jonathan2021/blog#4]
Let me just start by saying I really enjoyed this blog! It was a great resource to see how fast.ai code translates to raw PyTorch.
With that said, I ran the exact same code for 'fast.ai approach' portion in your blog, but I'm getting very poor performance of 0.2896 fastai_accuracy. I've ran it a couple of times but still getting below 40% accuracy.

![Screen Shot 2021-03-29 at 4 52 27 PM](https://user-images.githubusercontent.com/70506321/112804243-25a24480-90af-11eb-88f5-8df9867db4fe.png)

![Screen Shot 2021-03-29 at 4 54 16 PM](https://user-images.githubusercontent.com/70506321/112804452-669a5900-90af-11eb-81b4-bc68a102b0c7.png)

I attached two screenshots from my notebook; consider every other code the exact same as in your post. I would really appreciate if you can share your opinion on what may have happened here!

**Comment by Jonathan2021** [comment]
Hello @tylertaewook, thanks  for sharing ! Sorry for the late answer.
I ran the whole notebook again just in case I changed something without rerunning but I get the same results as in my blog. The only thing a bit out of the ordinary I did was working in a docker container using the [fastai docker image](https://github.com/fastai/docker-containers)
Things you can check:
- Your dataset is correctly constructed (Correct labels for each images). I see that your model really likes to predict 4, 8 and 9 (and got predicted correctly, especially for 4, more times than the other numbers) and that you don't have (or very few) predictions of the other numbers, even though they are present in your validation set (the rows in your confusion matrix, corresponding to the actual label, always have entries but not the columns).
- If your dataset is correctly downloaded and stored on your hard drive with the correct labels, it may be an issue with loading the dataset or on the train/validation set split (where training would only see 4s, 8s and 9s but the validation has other numbers).
- If you retrained your model several times without creating a new learner each time (the `cnn_learner(loaders, resnet34, metrics=accuracy)` cell), the model will behave as if countinuing training for many epochs and may overfit to the training data (but still it don't see how it could get that extreme). (I see that your loss on the training set converges nicely to 0 so the model is learning something).

Hope this will help you if you didn't fix the problem yet ! Good luck !
